s3path.py

```diff
@@ -244,6 +244,12 @@ def generate_prefix(self, path):
             return key_name + sep
         return key_name
 
+    def unlink(self, path, *args, **kwargs) -> None:
```

Comment:
This method are not raising FileNotFoundError if we have exceptions (for example if it's a bucket path without a key...)
So two options as I see it:
1. catch and raise FileNotFoundError exception
2. catch the relevant exception in the S3Path class method level...

Reply:
I'll catch the `ClientError` and raise the `FileNotFoundError` if that makes sense to you -- I think what you'll find is that the other approaches will wind up being very slow if you have to operate on many elements in the bucket for example (since each operation will require an API call). This way there is only one api call and we just need to deal with the failure

Based on how `pathlib` is structured it most likely makes sense to handle this in the `_S3Accessor` class. I'll add it there and see if that make ssense